src/Service/AwsS3/FolderSyncService.php

```
@@ -0,0 +1,84 @@
+use Psr\Log\LoggerInterface;
+
+class FolderSyncService
+{
+    private const FILESYSTEM_MOUNT_LOCAL_PREFIX = 'local';
+    private const FILESYSTEM_MOUNT_S3_PREFIX = 's3';
+
+    private Filesystem $filesystemLocal;
+    private Filesystem $filesystemS3;
+    private LoggerInterface $logger;
+    private string $awsS3Bucket;
+
+    public function __construct(
+        Filesystem $filesystemLocal,
+        Filesystem $filesystemS3,
+        string $awsS3Bucket,
+        LoggerInterface $logger
+    ) {
+        $this->filesystemLocal = $filesystemLocal;
+        $this->filesystemS3 = $filesystemS3;
+        $this->awsS3Bucket = $awsS3Bucket;
+        $this->logger = $logger;
+    }
+
+    public function copyUserFiles(): ?string
+    {
+        try {
+            $mountManager = new MountManager([
+                self::FILESYSTEM_MOUNT_LOCAL_PREFIX => $this->filesystemLocal,
+                self::FILESYSTEM_MOUNT_S3_PREFIX => $this->filesystemS3,
+            ]);
+
+            $contents = $this->filesystemLocal->listContents(date('Y-m-d'), true);
+            foreach ($contents as $item) {
+                if ('file' === $item['type']) {
+                    $sourcePath = sprintf('%s://%s', self::FILESYSTEM_MOUNT_LOCAL_PREFIX, $item['path']);
+                    $destinationPath = sprintf('%s://%s', self::FILESYSTEM_MOUNT_S3_PREFIX, $item['path']);
+
+                    if ($mountManager->has($destinationPath)) {
+                        $resource = $mountManager->readStream($sourcePath);
+                        if ($resource !== false) {
+                            $mountManager->putStream($destinationPath, $resource);
+                        }
+                    } else {
+                        $mountManager->copy($sourcePath, $destinationPath);
+                    }
+                }
+            }
+        } catch (S3Exception $exception) {
+            $this->logger->info($exception->getMessage());
+
+            return $exception->getMessage();
```

you don't need to return the message. you can throw a new exception right here. better a dedicated one. and catch it later where you want